nixos/modules/services/networking/wireguard.nix

```diff
@@ -23,8 +23,22 @@ let
 
       privateKey = mkOption {
         example = "yAnz5TF+lXXJte14tji3zlMNq+hd2rYUIgJBgB3fBmk=";
-        type = types.str;
-        description = "Base64 private key generated by wg genkey.";
+        type = with types; nullOr str;
+        default = null;
+        description = ''
+          DEPRECATED BECAUSE IT PUTS A WORLD-READABLE COPY OF THE KEY IN /nix/store!
```

Comment:
For some people, this might not be a problem. Rather than depreciating this, I'd keep both supported, but make them mutually exclusive.